### ISSUE TITLE
eth_getTransactionReceipt return "method handler crashed"

### ISSUE THREAD

**Issue by tjayrush** [ledgerwatch/turbo-geth#1790]
With a brand new fresh clone of commit `d7d790725ba2d70e8e5ce229feefb9d0af7065c7`, this curl command returns no data and 'method handler crashed.'

```
curl -s -H "Content-Type: application/json" -d '{"method":"eth_getTransactionReceipt","params":["0xc62d9410b73b18d767fe32df91e6336c5d563eceee49c55c56df65bfeb4b966d"],"id":1,"jsonrpc":"2.0"}' http://localhost:8545
```

call stack

```
ERROR[04-23|12:46:14.514] RPC method eth_getTransactionReceipt crashed: runtime error: invalid memory address or nil pointer dereference
goroutine 106 [running]:
github.com/ledgerwatch/turbo-geth/rpc.(*callback).call.func1(0xc000394980, 0x19, 0xc003303d50)
        github.com/ledgerwatch/turbo-geth/rpc/service.go:200 +0xbd
panic(0x10196c0, 0x1aa35f0)
        runtime/panic.go:965 +0x1b9
github.com/holiman/uint256.(*Int).ToBig(0x0, 0xc0000da000)
        github.com/holiman/uint256@v1.1.1/conversion.go:32 +0x55
github.com/ledgerwatch/turbo-geth/cmd/rpcdaemon/commands.marshalReceipt(0xc0032aa540, 0x12dbdf8, 0xc000238000, 0xc0003aaeb0)
        github.com/ledgerwatch/turbo-geth/cmd/rpcdaemon/commands/eth_receipts.go:282 +0x87
github.com/ledgerwatch/turbo-geth/cmd/rpcdaemon/commands.(*APIImpl).GetTransactionReceipt(0xc000142c80, 0x12cbe90, 0xc000143680, 0xd7183bb710942dc6, 0x6c33e691df32fe67, 0x5cc549eece3e565d, 0x6d964bebb
        github.com/ledgerwatch/turbo-geth/cmd/rpcdaemon/commands/eth_receipts.go:226 +0x37f
reflect.Value.call(0xc00012d000, 0xc000388528, 0x13, 0x1161b28, 0x4, 0xc0003aad70, 0x3, 0x3, 0xc0003aada0, 0xc0003a93f8, ...)
        reflect/value.go:476 +0x8e7
reflect.Value.Call(0xc00012d000, 0xc000388528, 0x13, 0xc0003aad70, 0x3, 0x3, 0x0, 0x0, 0x10efda0)
        reflect/value.go:337 +0xb9
github.com/ledgerwatch/turbo-geth/rpc.(*callback).call(0xc0003878c0, 0x12cbe90, 0xc000143680, 0xc000394980, 0x19, 0xc0003a93f8, 0x1, 0x1, 0x0, 0x0, ...)
        github.com/ledgerwatch/turbo-geth/rpc/service.go:206 +0x2c5
github.com/ledgerwatch/turbo-geth/rpc.(*handler).runMethod(0xc0001b4d80, 0x12cbe90, 0xc000143680, 0xc00034d1f0, 0xc0003878c0, 0xc0003a93f8, 0x1, 0x1, 0x1)
        github.com/ledgerwatch/turbo-geth/rpc/handler.go:399 +0x8a
github.com/ledgerwatch/turbo-geth/rpc.(*handler).handleCall(0xc0001b4d80, 0xc000216f60, 0xc00034d1f0, 0xc0014056c0)
        github.com/ledgerwatch/turbo-geth/rpc/handler.go:347 +0x252
github.com/ledgerwatch/turbo-geth/rpc.(*handler).handleCallMsg(0xc0001b4d80, 0xc000216f60, 0xc00034d1f0, 0x12c0601)
        github.com/ledgerwatch/turbo-geth/rpc/handler.go:301 +0x1be
github.com/ledgerwatch/turbo-geth/rpc.(*handler).handleMsg.func1(0xc000216f60)
        github.com/ledgerwatch/turbo-geth/rpc/handler.go:142 +0x46
github.com/ledgerwatch/turbo-geth/rpc.(*handler).startCallProc.func1(0xc0001b4d80, 0xc0003a93e0)
        github.com/ledgerwatch/turbo-geth/rpc/handler.go:229 +0xd2
created by github.com/ledgerwatch/turbo-geth/rpc.(*handler).startCallProc
        github.com/ledgerwatch/turbo-geth/rpc/handler.go:225 +0x66
```

I'm going to try to fix this, as I need it for my testing.

**Comment by tjayrush** [comment]
Looking more closely at the code. This line https://github.com/ledgerwatch/turbo-geth/blob/master/cmd/rpcdaemon/commands/eth_receipts.go#L271 create chainId which is not initialized and at this line https://github.com/ledgerwatch/turbo-geth/blob/master/cmd/rpcdaemon/commands/eth_receipts.go#L282 it crashed because `chainId` is <nil>.

**Comment by MysticRyuujin** [comment]
```
$ curl -s -H "Content-Type: application/json" -d '{"method":"eth_getTransactionReceipt","params":["0xc62d9410b73b18d767fe32df91e6336c5d563eceee49c55c56df65bfeb4b966d"],"id":1,"jsonrpc":"2.0"}' http://localhost:8544

{"jsonrpc":"2.0","id":1,"result":{"blockHash":"0x0f1217b92276cd17608d4212879739e6a5ec388bd7a03bef9798655234afd2b9","blockNumber":"0x37cf08","contractAddress":null,"cumulativeGasUsed":"0x14820","from":"0x64014a5817e05a205bca161a6763e945659b50b0","gasUsed":"0x5208","logs":[],"logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","status":"0x1","to":"0x0699c7874bc28fa9d8b1a17e4dd385fe5aa17614","transactionHash":"0xc62d9410b73b18d767fe32df91e6336c5d563eceee49c55c56df65bfeb4b966d","transactionIndex":"0x3","type":"0x0"}}
```


```
$ git show
commit e25b57511df88c7315e5964a7217a29302861418 (HEAD -> master, origin/master, origin/HEAD)
Author: alex.sharov <AskAlexSharov@gmail.com>
Date:   Tue Apr 20 23:34:54 2021 +0700

    fix nil pointer in peer
```

**Comment by tjayrush** [comment]
Closed by #1792
